src/components/inputs/z-input/index.tsx

```diff
@@ -294,25 +267,45 @@ export class ZInput {
       <z-icon
         class="resetIcon"
         name="multiply"
-        onClick={(e: any) => this.emitInputChange("", e.keyCode)}
+        onClick={() => this.emitInputChange("")}
+        onKeyUp={(e: KeyboardEvent) =>
+          handleKeyboardSubmit(e, this.emitInputChange.bind(this), "")
+        }
+        role="button"
+        tabIndex={0}
+        aria-label="cancella il contenuto dell'input"
       />
     );
   }
 
   renderShowHidePassword() {
+    const togglePassword = () => (this.passwordHidden = !this.passwordHidden);
     return (
       <z-icon
         class="showHidePasswordIcon"
         name={this.passwordHidden ? "view-filled" : "view-off-filled"}
-        onClick={() => (this.passwordHidden = !this.passwordHidden)}
+        onClick={() => togglePassword()}
+        onKeyUp={(e: KeyboardEvent) => handleKeyboardSubmit(e, togglePassword)}
+        role="button"
+        tabIndex={this.disabled ? -1 : 0}
+        aria-label={
+          this.passwordHidden ? "mostra password" : "nascondi password"
+        }
       />
     );
   }
 
   renderMessage() {
-    if (!this.hasmessage) return;
+    if (boolean(this.message) === false) return;
 
-    return <z-input-message message={this.message} status={this.status} />;
+    return (
+      <z-input-message
+        message={
+          boolean(this.message) === true ? undefined : (this.message as string)
```

Comment:
Il controllo su `boolean(this.message)` è già presente a riga 299, no? La condizione poi qui non dovrebbe essere al contrario?

Reply:
A linea 299 controllo solo se è `false`, in quel caso devo proprio uscire.
Se invece è `true`, lo devo renderizzare vuoto (e non devo renderizzare stringa "true"), in tutti gli altri casi passi diretta la stringa `this.message`

Reply:
vero, scusami